packages/tracker/tracker.js

```diff
@@ -578,6 +578,37 @@ Tracker.autorun = function (f, options) {
   return c;
 };
 
+/**
+ * @callback Tracker.ComputationFunction
+ * @param {Tracker.Computation}
+ */
+/**
+ * @summary Await-able Tracker.autorun. Run a function now and rerun it later whenever its dependencies
+ * change. Returns a promise for the Computation object that can be used to stop or observe the
+ * rerunning.
+ * @locus Client
+ * @param {Tracker.ComputationFunction} runFunc The function to run. It receives
+ * one argument: the Computation object that will be returned.
+ * @param {Object} [options]
+ * @param {Function} options.onError Optional. The function to run when an error
+ * happens in the Computation. The only argument it receives is the Error
+ * thrown. Defaults to the error being logged to the console.
+ * @returns {Tracker.Computation}
+ */
+Tracker.autorunAsync = function (f, options) {
+  let fPromise
+  const newF = async(...args) => {
+    fPromise = f(...args)
+  }
+
+  const c = Tracker.autorun(newF, options)
+
+  return fPromise.then(() => {
+    return c
+  })
```

Comment:
While this implementation is good (the names could be improved and the `newF` does not have to be `async`), I have a suggestion: instead of adding a new `Tracker` function (and `Template` function in https://github.com/meteor/blaze/pull/427), we could add something like `computation.firstRunPromise`.

The usage is fairly similar: instead of `await Tracker.autorunAsync()` you'd do `await Tracker.autorun().firstRunPromise` (eventually we could add `autorunAsync` as a convenience wrapper).

The main difference to me is that with the current implementation it's impossible to stop this computation before it finishes the first round. Of course, we won't implement cancellation just yet (maybe in the future?) but the `.stop()` function is simply not there until the first call finishes.

Reply:
Thank you @radekmie I'll try that, that sounds like an improvement, which I'm all for! :D 

> the names could be improved

Ah that hurt! 😄